cfme/utils/template/base.py

```diff
@@ -184,13 +186,37 @@ def ssh_client_args(self):
                 'username': credentials[cred_key]['username'],
                 'password': credentials[cred_key]['password']}
 
+    @cached_property
+    def raw_vm_ssh_client_args(self):
+        """ Returns credentials + hostname for ssh client auth for a temp_vm."""
+
+        def check_ip():
+            vm_name = (self.temp_vm_name if self.provider_data.type == 'rhevm' else
+                       self.template_name)
+            vm = self.mgmt.get_vm(vm_name)
+            ip = vm.ip
+            # get_ip_address might return None
+            return ip if ip and resolve_hostname(ip) else False
+
+        vm_ip, tc = wait_for(check_ip, delay=5, num_sec=600,
+                             message="waiting for {} to obtain DHCP"
+                             .format(self.temp_vm_name))
+
+        return {'hostname': vm_ip,
+                'username': credentials['ssh']['username'],
+                'password': credentials['ssh']['password']}
+
     @cached_property
     def tool_client_args(self):
         tool_data = self.from_template_upload('tool_client')
         return {'hostname': tool_data.hostname,
             'username': credentials[tool_data['credentials']].username,
             'password': credentials[tool_data['credentials']].password}
 
+    @cached_property
+    def _vm_mgmt(self):
+        return self.mgmt.get_vm(self.template_name)
```

Comment:
This ends up being a bit ambiguous as a property.  According to the logic in `raw_vm_ssh_client_args`, this name could depend on the provider type?

Without comments here, its not clear why its different for rhevm in some cases, but not others.

Is this vm_mgmt for the temp VM? for the raw VM?